app/javascript/components/search/results/StudySearchResult.jsx

```diff
@@ -170,17 +175,109 @@ function studyTypeBadge(study) {
   }
 }
 
+/**
+ * Log analytics about search result selection to Mixpanel
+ *
+ * This logging differentiates properties about the selected result from
+ * properties about other aspects of the search -- e.g. the full result set
+ * and query -- by prefixing the latter with a "namespace", i.e. `results:`.
+ *
+ * @param {Object} study Selected result
+ * @param {Object} logProps Other properties to log
+ *
+ * TODO (SCP-4256): Refine logged properties for search result selection
+ * TODO (SCP-4257): Eliminate duplicate default/custom search event logging
+ */
+export function logSelectSearchResult(study, logProps={}) {
+  logProps = Object.assign(
+    {
+      studyAccession: study.accession,
+      context: 'study',
+      scope: 'global'
+    },
+    study,
+    logProps
+  )
+  delete logProps.accession // Redundant with more-conventional studyAccession
+
+  Object.entries(logProps.results).forEach(([key, value]) => {
+    if (['termList', 'geneList'].includes(key)) {
+      // Redundant with pre-existing `terms` and `genes` Mixpanel props
+      return
+    }
+    logProps[`results:${key}`] = value
+  })
+  delete logProps.results // Remove nested property; flattened above
+
+  // Number of searches done before this result was selected.
+  // This is needed to easily analyze changes in click-through rate (CTR).
+  logProps.numSearches = numSearches
+
+  // We don't log study name, like Terra doesn't log workspace name, per
+  // original DSP Mixpanel design doc
+  delete logProps.name
+  delete logProps.study_url // Has name
+  delete logProps.description // Too similar to name
+  delete logProps['results:studies'] // Includes above
+
+  // Objects can't be queried in Mixpanel, so flatten matchByData
+  if (logProps['results:matchByData'] !== null) {
+    Object.entries(logProps['results:matchByData']).forEach(([key, value]) => {
+      logProps[`results:${key}`] = value
+    })
+  }
+  delete logProps['results:matchByData']
+
+  let refinedLogProps = {}
+  Object.entries(logProps).forEach(([key, value]) => {
+    // Counts of "foo" in Mixpanel are conventionally structured as `numFoos`.
```

Comment:
there's only 3 of these properties that I see, handling them manually will make this code less error-prone.

Reply:
Done in 8d57e81.  It feels less robust, but more understandable.